covsirphy/cleaning/jhu_data.py

```diff
@@ -591,30 +588,39 @@ def show_complement(self, country=None, province=None,
                     - Full_recovered (bool): True if applied for recovered or False otherwise
                     - Partial_recovered (bool): True if applied for recovered or False otherwise
         """
-        complement_df = pd.DataFrame(columns=[self.COUNTRY, self.PROVINCE,
-                                              *JHUDataComplementHandler.SHOW_COMPLEMENT_FULL_COLS])
-        complement_df.set_index(self.COUNTRY, inplace=True)
         self._recovery_period = self._recovery_period or self.calculate_recovery_period()
+        # Area name
         if country is None:
-            country = [c for c in self._cleaned_df[self.COUNTRY].unique() if c != "Others"]
+            country = [
+                c for c in self._cleaned_df[self.COUNTRY].unique() if c != "Others"]
         province = province or self.UNKNOWN
         if not isinstance(country, str) and province != self.UNKNOWN:
-            raise ValueError("@province cannot be specified when @country is not a string.")
+            raise ValueError(
+                "@province cannot be specified when @country is not a string.")
         if not isinstance(country, list):
             country = [country]
+        # Create complement handler
+        handler = JHUDataComplementHandler(
+            recovery_period=self._recovery_period,
+            interval=interval,
+            max_ignored=max_ignored,
+            max_ending_unupdated=max_ending_unupdated,
+        )
```

Comment:
Are you sure that the handler `JHUDataComplementHandler` object does not need to be reinitialized during each iteration?

Reply:
I think it is not required because
- the sets of arugments (`recovery_period, `interval`, `max_ignored`, `max_ending_unupdated`) are the same for all countries and
- `self.complement_dict` will be initialized by `dict.fromkeys(self.SHOW_COMPLEMENT_FULL_COLS, False)` in  `.run().

Is not correct?